src/channels.js

```diff
@@ -55,13 +55,15 @@ const updateChannels = (app, user) => {
         }
     });
 
-   // assert(connections.length == 1);
+    // assert(connections.length == 1);
 
-    // Leave all channels.
-    leaveChannels(app, user);
-    
-    // Re-join all channels with the updated user information.
-    joinChannels(app, user, connections[0]);
+    if (connections.length === 1) {
```

Comment:
Why do we need this? 

I.e.: What does not work properly with this code and why.

Reply:
Sometimes (it's rare) connections are empty and then `joinChannels` function throws an error and server is killed because there're no connections to leave and then re-join. This change prevent this behaviour.

Reply:
@kaniak274 Ok got it. It looks like a useful thing to know to please move it to the comment in the code.